data_structures/src/chain.rs

```diff
@@ -238,6 +241,32 @@ pub struct Transaction {
 }
 
 impl Transaction {
+    /// Creates a new transaction from inputs and outputs.
+    // TODO: Transaction::new is missing the signatures which depend
+    // on a sign key that needs to be implemented first in
+    // witnet_crypto.
+    pub fn new(inputs: Vec<Input>, outputs: Vec<Output>, _sign_key: ()) -> Self {
+        let mut signatures = Vec::with_capacity(inputs.len());
+
+        for _ in inputs.iter() {
+            signatures.push(KeyedSignature {
+                signature: Signature::Secp256k1(Secp256k1Signature {
+                    r: [0; 32],
+                    s: [0; 32],
+                    v: 0,
+                }),
+                public_key: [0; 32],
+            })
+        }
```

Comment:
Why not functional? :stuck_out_tongue_closed_eyes: 

```rust
let signatures = inputs.iter()
    .map(|_input| KeyedSignature {
        signature: Signature::Secp256k1(Secp256k1Signature {
            r: [0; 32],
            s: [0; 32],
            v: 0,
        }),
        public_key: [0; 32],
    })
    .collect();
```

Reply:
I was worried that `.collect()` didn't create a vector of the right capacity and thus were a bit less performant, but I was wrong, [this is how it is implemented](https://doc.rust-lang.org/src/alloc/vec.rs.html#1782)